tools/xtce_generation/telemetry_generator.py

```diff
@@ -0,0 +1,341 @@
+
+        for parameter_data in ccsds_parameters:
+            parameter_ref_entry = Et.SubElement(
+                ccsds_packet_entry_list, "xtce:ParameterRefEntry"
+            )
+            parameter_ref_entry.attrib["parameterRef"] = parameter_data["name"]
+
+        # Create CoDICESciencePacket SequenceContainer
+        science_container = Et.SubElement(container_set, "xtce:SequenceContainer")
+        science_container.attrib["name"] = "CoDICESciencePacket"
+
+        base_container = Et.SubElement(science_container, "xtce:BaseContainer")
+        base_container.attrib["containerRef"] = "CCSDSPacket"
+
+        restriction_criteria = Et.SubElement(base_container, "xtce:RestrictionCriteria")
+        comparison = Et.SubElement(restriction_criteria, "xtce:Comparison")
+        comparison.attrib["parameterRef"] = "PKT_APID"
+        comparison.attrib["value"] = apid
+        comparison.attrib["useCalibratedValue"] = "false"
+
+        Et.SubElement(science_container, "xtce:EntryList")
+
+        return telemetry_metadata
+
+    def add_science_parameters(self, science_container, pkt):
+        """
+        Add ParameterRefEntry elements for SciencePacket.
+
+        Parameters:
+        - science_container: The SciencePacket SequenceContainer element.
+        - pkt: The DataFrame containing packet data.
+
+        Returns:
+        - codice_science_container: The updated SciencePacket
+        SequenceContainer element.
+        """
+        # Get the 'mnemonic' values starting from row 10
+        parameter_refs = pkt.loc[9:, "mnemonic"].tolist()
+
+        for parameter_ref in parameter_refs:
+            parameter_ref_entry = Et.SubElement(
+                science_container, "xtce:ParameterRefEntry"
+            )
+            parameter_ref_entry.attrib["parameterRef"] = parameter_ref
+
+        return science_container
+
+    def create_parameters_from_dataframe(self, parameter_set, pkt):
+        """
+        Create XML elements for parameters based on DataFrame rows starting from row 10.
+
+        Parameters:
+        - parameter_set: The ParameterSet element where parameters will be added.
+        - pkt: The DataFrame containing packet data.
+
+        Returns:
+        - parameter_set: The updated ParameterSet element.
+        """
+        # Process rows from 10 until the last available row in the DataFrame
+        for index, row in pkt.iterrows():
+            if index < 9:
+                continue  # Skip rows before row 10
+
+            parameter = Et.SubElement(
+                parameter_set, "{http://www.omg.org/space}Parameter"
+            )
+            parameter.attrib["name"] = row["mnemonic"]
+            parameter_type_ref = ""
+
+            # Use BYTE type for "Data" and "Event_Data" mnemonics
+            if row["mnemonic"] in ["Data", "Event_Data"]:
+                parameter_type_ref = "BYTE"
+            else:
+                parameter_type_ref = f"uint{row['lengthInBits']}"  # Use UINT for others
```

Comment:
```suggestion
                parameter_type_ref = f"uint{int(row['lengthInBits'])}"  # Use UINT for others
```

Same as above. If it would be possible to get this value from the `unique_lengths` variable above, it's better to only parse through the dataframe once exactly because of these type issues. But, it doesn't matter for this review - just wanted to note that for the future.